src/main/java/hudson/plugins/cigame/rules/plugins/findbugs/AbstractFindBugsWarningsRule.java

```diff
@@ -55,18 +64,33 @@ protected int getNumberOfAnnotations(List<FindBugsResultAction> list) {
     	}
     	
     	List<FindBugsResultAction> currentActions = ActionRetriever.getResult(build, Result.UNSTABLE, FindBugsResultAction.class);
-    	if (!hasNoErrors(currentActions)) {
-    		return EMPTY_RESULT;
-    	}
-    	int currentAnnotations = getNumberOfAnnotations(currentActions);
-    		
-    	List<FindBugsResultAction> previousActions = ActionRetriever.getResult(previousBuild, Result.UNSTABLE, FindBugsResultAction.class);
-    	if (!hasNoErrors(previousActions)) {
-    		return EMPTY_RESULT;
-    	}
-    	int previousAnnotations =getNumberOfAnnotations(previousActions);
-    	
-    	return evaluate(previousAnnotations, currentAnnotations);
+		if (currentActions.isEmpty()) {
+			return evaluateMaven(previousBuild, build);
+		} else {
+			if (!hasNoErrors(currentActions)) {
+				return EMPTY_RESULT;
+			}
+			int currentAnnotations = getNumberOfAnnotations(currentActions);
+			
+			List<FindBugsResultAction> previousActions = ActionRetriever.getResult(previousBuild, Result.UNSTABLE, FindBugsResultAction.class);
+			if (!hasNoErrors(previousActions)) {
+				return EMPTY_RESULT;
+			}
+			int previousAnnotations = getNumberOfAnnotations(previousActions);
+			
+			return evaluate(previousAnnotations, currentAnnotations);
+		}
+	}
+
+	private RuleResult<Integer> evaluateMaven(AbstractBuild<?, ?> previousBuild, AbstractBuild<?, ?> build) {
```

Comment:
It seems like there should be at first glance but I couldn't get that to work.  If I remember correctly, my first attempt at reworking this method was to rewrite the getNumberOfAnnotations method to be generic, but the FindBugsResultAction.getResult() method and the FindBugsMavenResultAction.getResult() method do not belong to the same super class so that didn't work.

The other big difference between the else block in the evaluate(AbstractBuild, AbstractBuild) method and the body of the evaluateMaven(AbstractBuild, AbstractBuild) method is the omission of the "hasNoErrors" checks.  I purposefully left these out of the evaluateMaven method because I found that the Maven warning "File encoding has not been set in pom.xml, using platform encoding UTF-8, i.e. build is platform dependent" was being reported as an error.